deployment_report/model/utils/config_util.py

```diff
@@ -54,66 +73,175 @@ def get_cadence_version(config_maps: Dict) -> Optional[Text]:
              ConfigMap does not contain the requested information.
     """
     # look for the necessary ConfigMap
-    for config_map_name, config_map_details in config_maps.items():
-        if _SAS_DEPLOYMENT_METADATA_CONFIG_MAP_NAME_ in config_map_name:
-            # get the KubernetesResource
-            config_map: KubernetesResource = \
-                config_map_details[ViyaDeploymentReportKeys.ResourceDetails.RESOURCE_DEFINITION]
 
-            # get the ConfigMap data
-            cadence_data: Optional[Dict] = config_map.get_data()
+    try:
+        if not resource:
+            return None
 
-            try:
-                # build the cadence string
-                return (
-                    f"{cadence_data[_CADENCE_DISPLAY_NAME_KEY_]} "
-                    f"{cadence_data[_CADENCE_VERSION_KEY_]} "
-                    f"({cadence_data[_CADENCE_RELEASE_KEY_]})"
-                )
-            except KeyError:
-                # if an expected key wasn't defined, continue
-                continue
+        config_maps: Dict = resource[ResourceTypeValues.K8S_CORE_CONFIG_MAPS][ITEMS_KEY]
+
+        for config_map_name, config_map_details in config_maps.items():
+            if _SAS_DEPLOYMENT_METADATA_CONFIG_MAP_NAME_ in config_map_name:
+                # get the KubernetesResource
+                config_map: KubernetesResource = \
+                    config_map_details[Keys.ResourceDetails.RESOURCE_DEFINITION]
+
+                # get the ConfigMap data
+                cadence_data: Optional[Dict] = config_map.get_data()
 
-    # expected value wasn't found
-    return None
+                try:
+                    # build the cadence string
+                    return (
+                        f"{cadence_data[_CADENCE_DISPLAY_NAME_KEY_]} "
+                        f"{cadence_data[_CADENCE_VERSION_KEY_]} "
+                        f"({cadence_data[_CADENCE_RELEASE_KEY_]})"
+                    )
+                except KeyError:
+                    # if an expected key wasn't defined, continue
+                    return ("Version information is not defined")
 
+        # expected value wasn't found
+        return None
 
-def get_db_info(config_maps: Dict) -> Dict:
+    except KeyError:
+        return None
+
+
+def get_db_info(resource: Dict) -> Dict:
     """
     Returns the db information of the targeted SAS deployment.
 
-    :param config_maps: The ConfigMap resources to evaluate.
+    :param resource: The api resource data.
+
+    :return: A dict representing the db information of the targeted SAS deployment, or None if
+             no resource information
```

Comment:
I think this sentence needs to be completed, something like "...if no resource information was cached."